gc/src/gc.rs

```diff
@@ -52,6 +52,7 @@ const ROOTS_MAX: usize = ROOTS_MASK; // max allowed value of roots
 pub(crate) struct GcBoxHeader {
     roots: Cell<usize>, // high bit is used as mark flag
     next: Cell<Option<NonNull<GcBox<dyn Trace>>>>,
+    dyn_trace_ptr: Cell<Option<NonNull<GcBox<dyn Trace>>>>,
```

Comment:
If `NonNull<GcBox<dyn Trace>>`'s are what it makes sense to put in the "queue" then I wasn't clear on how to get ahold of one at the time of calling `Gc`'s `Trace::trace()` implementation (I seemed to be running into the fact that since there `T` can be `?Sized` the compiler is saying it doesn't know how to make a trait object/fat pointer for something it doesn't know the size of or something along those lines?) so I resorted to storing one "up front" at the time of creation of `Gc`/`GcBox`/`GcBoxHeader`

Maybe another option to circumvent that would be to push closures (eg `Box<dyn Fn()>`) onto a queue instead of `NonNull<GcBox<dyn Trace>>`'s?